Sources/Vitamin/Components/Tag/VitaminTag.swift

```diff
@@ -0,0 +1,147 @@
+//
+//  Vitamin iOS
+//  Apache License 2.0
+//
+
+import UIKit
+
+public class VitaminTag: UILabel {
+    // padding constants of the `VitaminTag`
+    private let padding = UIEdgeInsets(top: 2, left: 8, bottom: 2, right: 8)
+
+    /// Label of the `VitaminTag`
+    public var label: String = "" {
+        didSet {
+            applyTextAndIcon()
+        }
+    }
+
+    /// Icon displayed on the `VitaminTag`
+    public var icon: UIImage? {
+        didSet {
+            applyTextAndIcon()
+        }
+    }
+
+    /// Variant of the `VitaminTag`
+    public var variant: Variant = .brandPrimary {
+        didSet {
+            self.layer.backgroundColor = variant.backgroundColor.cgColor
+            self.textColor = variant.foregroundColor
+        }
+    }
+
+    /// An initializer that restores a `VitaminTag` from a serialized version (used in storyboard)
+    public required init?(coder aDecoder: NSCoder) {
+        super.init(coder: aDecoder)
+        commonInit()
+    }
+
+    /// An initializer that instantiate a `VitaminTag` in a frame, with all default valmues for properties
+    override public init(frame: CGRect) {
+        super.init(frame: frame)
+        commonInit()
+    }
+
+    // Common initialization
+    private func commonInit() {
+        self.backgroundColor = .clear
+        self.layer.cornerRadius = VitaminRadius.radius100.rawValue
+    }
+}
+
+// MARK: - Padding methods
+extension VitaminTag {
+    override public func drawText(in rect: CGRect) {
+        super.drawText(in: rect.inset(by: padding))
+    }
+
+    override public var intrinsicContentSize: CGSize {
+        let superContentSize = super.intrinsicContentSize
+        let width = superContentSize.width + padding.left + padding.right
+        let heigth = superContentSize.height + padding.top + padding.bottom
+        return CGSize(width: width, height: heigth)
+    }
+}
+
+// MARK: - Tag Variant
+extension VitaminTag {
+    /// enum containing the different variants of VitaminTag
+    public enum Variant {
```

Comment:
ok for the first option.
Because, the second one would not in my understanding be a solution, because, we still should link these new colors with the différent Variant.
And these new colors would be duplicate of already existing semantic colors...